airbyte_cdk/sources/declarative/retrievers/simple_retriever.py

```diff
@@ -626,6 +627,102 @@ def _to_partition_key(to_serialize: Any) -> str:
+            records_schema: JSON schema describing the record structure
+
+        Returns:
+            The fetched record as a dict.
+
+        Raises:
+            RecordNotFoundException: If the response is empty/ignored or parsing yields no records.
+            Exception: HTTP errors (including 404) are propagated from requester's error handling.
+
+        Example:
+            record = retriever.fetch_one("123", schema)
+
+        Note:
+            This implementation uses convention-based path construction, appending /{pk_value} to the base path. (important-comment)
+
+            Alternative approaches that could be implemented in the future: (important-comment)
+            - Path template interpolation: Use a configurable template like "{base_path}/{id}" (important-comment)
+              See: https://github.com/airbytehq/airbyte-python-cdk/issues/833#phase-1a (important-comment)
+            - Field path configuration: Allow specifying which response field contains the record (important-comment)
+              for APIs that wrap single records in envelopes like {"data": {...}} (important-comment)
+        """
+        # Single-record fetch doesn't involve partitioning, so we pass an empty StreamSlice
+        empty_stream_slice = StreamSlice(partition={}, cursor_slice={})
+
+        # Get the base path from the requester
+        base_path = self.requester.get_path(
+            stream_state={},
+            stream_slice=empty_stream_slice,
+            next_page_token=None,
+        )
+
+        fetch_path = f"{base_path.rstrip('/')}/{pk_value.lstrip('/')}"
+
+        # send_request() may return None when the error handler chooses to IGNORE a response
+        response: requests.Response | None = self.requester.send_request(
+            path=fetch_path,
+            stream_state={},
+            stream_slice=empty_stream_slice,
+            next_page_token=None,
+            request_headers=self._request_headers(
+                stream_state={},
+                stream_slice=empty_stream_slice,
+                next_page_token=None,
+            ),
+            request_params=self._request_params(
+                stream_state={},
+                stream_slice=empty_stream_slice,
+                next_page_token=None,
+            ),
+            request_body_data=self._request_body_data(
+                stream_state={},
+                stream_slice=empty_stream_slice,
+                next_page_token=None,
+            ),
+            request_body_json=self._request_body_json(
+                stream_state={},
+                stream_slice=empty_stream_slice,
+                next_page_token=None,
+            ),
+            log_formatter=self.log_formatter,
+        )
+
+        if not response:
+            raise RecordNotFoundException(
+                f"Record with primary key {pk_value} not found (no response)"
+            )
+
+        records_iter: Iterable[Record] = self._parse_response(
```

Comment:
This section doesn't seem like it will work for a lot of cases. I feel like a common pattern is that for retrieving all objects like `prices`, we get back a response w/ a top-level array object which will contain an object for each price. And when we hit the individual API endpoint we get back a single top-level object representing the price.

The way this is written assumes that hitting the single record `fetch_path` will contain a top level list and extract each record into the `records_iter`. But now we'll be unable to extract anything, there won't be a `first_record` to extract.

Unless you think I'm missing something, it seems like we shouldn't be using the underlying extractor which is no longer suitable for individual records